src/Microsoft.OData.Core/UriParser/Parsers/FunctionCallParser.cs

```diff
@@ -60,11 +77,15 @@ public ExpressionLexer Lexer
         /// Parses an identifier that represents a function.
         /// </summary>
         /// <param name="parent">Token for the parent of the function being parsed.</param>
-        /// <returns>QueryToken representing this function.</returns>
-        public QueryToken ParseIdentifierAsFunction(QueryToken parent)
+        /// <param name="result">QueryToken representing this function.</param>
+        /// <returns>True if the parsing was successful.</returns>
+        public bool ParseIdentifierAsFunction(QueryToken parent, out QueryToken result)

```

Comment:
You have essentially made a TryParse... method without the naming of one.  I'd suggest a naming change.  It looks like there is just one caller in the product side so I think changing the signature is fine (otherwise, I'd suggest a TryParse... and a Parse... method).